packages/eui/src/components/code/code_block.test.tsx

```diff
@@ -174,40 +130,38 @@ describe('EuiCodeBlock', () => {
           {code}
         </EuiCodeBlock>
       );
-      expect(container.firstChild).toMatchSnapshot();
+
+      expect(container).toBeInTheDocument();
```

Comment:
Depending on the size of the snapshot, I wouldn't hate this being left as a snapshot - it's not bad to get an idea of what kind of DOM our virtualization library renders and what inline styles it's adding.

If it's a crapton of lines that would cause people's eyes to glaze over however, I'm good with leaving it as an assertion:

```suggestion
      expect(container.querySelector('[class*="euiCodeBlock__code-isVirtualized"]')).toBeInTheDocument();
```

Reply:
> Depending on the size of the snapshot, I wouldn't hate this being left as a snapshot - it's not bad to get an idea of what kind of DOM our virtualization library renders and what inline styles it's adding.

That's what I thought initially but after revisiting our discussion about snapshot tests, reading different opinions and reflecting on it myself, I came to a conclusion that testing the component structure is too prone to misinterpretation and human error while deciding if the diff is expected or not. It's easy to skip by just updating the snapshot. The core aspects we want to test (like rendering, important classes, or roles) can be tested with proper assertions.

Still, the suggestion you made ☝🏻 I don't like that we have to query by a class like that but there's no other way I see in a unit test and I do like that the class points more to whether the virtualization is applied or not. BUT I believe the better way to test it is with Cypress, so making sure only the visible items are rendered with `EuiCodeBlock` that has a `isVirtualized` prop set to `true`.

Let me know what you think and we decide together how it's better to test virtualization.